types/underscore/underscore-tests.ts

```diff
@@ -152,187 +152,205 @@ declare const maybeFunction: (() => void) | undefined;
-_.min(numbers);
+// retrieving the item with the maximum number in a property
+_.max(stooges, (stooge) => stooge.age);
 
+// sorting by a calculated value
 _.sortBy([1, 2, 3, 4, 5, 6], (num) => Math.sin(num));
 
-_([1, 2, 3]).chain()
-    .sortBy(x => -x)
-    .sortBy(x => -x)
-    .value().length;
-
+// grouping numbers by their non-fractional parts
 _([1.3, 2.1, 2.4]).groupBy((e) => Math.floor(e));
-_.groupBy([1.3, 2.1, 2.4], (num) => Math.floor(num).toString());
-_.groupBy(['one', 'two', 'three'], 'length');
-
-_.indexBy(stooges, 'age')['40'].age;
-_(stooges).indexBy('age')['40'].name;
-_(stooges)
-    .chain()
-    .indexBy('age')
-    .value()['40'].age;
 
-let pensioners: string[] = _.chain(stooges)
-    .filter(p => p.age >= 60)
-    .map(p => p.name)
-    .value();
-
-var usersData: _.Dictionary<{ age: number; name: string }> = {
-    'user id': { name: 'moe', age: 40 },
-    'other user Id': { name: 'larry', age: 50 },
-    'fake id': { name: 'curly', age: 60 },
-};
-
-let youngPeopleId: string[] = _.chain(usersData)
-    .map((p, k: string) => k)
-    .value();
-
-let usersTable: { age: number; name: string; id: string }[] = _.chain(usersData)
-    .map((p, k: string) => {
-        return { id: k, ...p };
-    })
-    .value();
-
-// Test map function with _ChainOfArrays<>
-let usersTable_2 /*: { age: number; name: string; id: string }[][]*/ = _.chain(usersData)
-    .map((p, k: string) => {
-        return [{ id: k, ...p }];
-    })
-    .value();
+// grouping numbers by the value of a specified property
+_.groupBy(['one', 'two', 'three'], 'length');
 
-let usersTable_3 /*: { score: number; fullName: string; login: string }[][]*/ = _.chain(usersTable)
-    .map(p => {
-        return [
-            {
-                login: p.id,
-                fullName: p.name,
-                score: p.age,
-            },
-        ];
-    })
-    .value();
+// indexing items in a dictionary by age
+_.indexBy(stooges, 'age');
 
+// counting numbers by their evenness
 _.countBy([1, 2, 3, 4, 5], (num) => (num % 2 == 0) ? 'even' : 'odd');
 
+// shuffling numbers
 _.shuffle([1, 2, 3, 4, 5, 6]);
 
+// converting an array-like structure to an array
 (function (a, b, c, d) { return _.toArray(arguments).slice(1); })(1, 2, 3, 4);
 
+// determining the number of items in a dictionary
 _.size({ one: 1, two: 2, three: 3 });
 
+// splitting numbers into sets of even and odd values
 _.partition<number[]>([0, 1, 2, 3, 4, 5], (num) => {return num % 2 == 0 });
 
 interface Family {
     name: string;
     relation: string;
 }
+
+// creating a function that can determine if one object's property values matches another's
 var isUncleMoe = _.matches<Family>({ name: 'moe', relation: 'uncle' });
-_.filter([{ name: 'larry', relation: 'father' }, { name: 'moe', relation: 'uncle' }], isUncleMoe);
 var uncleMoe: Family = { name: 'moe', relation: 'uncle' };
 isUncleMoe(uncleMoe);
 
 ///////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Any idea what these dividers are for? Might be worth documenting those as well.

Reply:
These were roughly separators between the different Underscore function areas (Collections, Arrays, Functions, Objects, Utility) and almost all tests are within their correct group. I was originally going to replace these with section headers like https://github.com/DefinitelyTyped/DefinitelyTyped/compare/master...reubenrybnik:underscore-tests-usage-tests-cleanup-old#diff-4d35bcc89b6520b7b1930f73c7e9c4bfR365, but since the revised plan is to move these tests down with their corresponding combinatorial tests I decided not to bother changing the dividers up here since those tests already have both group- and function-level headers.